Source/Core/VideoCommon/HiresTextures.cpp

```diff
@@ -184,34 +235,50 @@ void HiresTexture::Prefetch()
       }
       if (iter != s_textureCache.end())
       {
-        for (const Level& l : iter->second->m_levels)
+        for (const Level& level : iter->second->m_levels)
         {
-          size_sum += l.data_size;
+          prefetched_size_sum += level.data_size;
+          // a texture without mipmaps is considered to have 1 mip level
         }
       }
     }
 
-    if (s_textureCacheAbortLoading.IsSet())
+    if (s_textureCacheAbortPrefetching.IsSet())
     {
       return;
     }
 
-    if (size_sum > max_mem)
+    // Stops prefetching to avoid getting out of RAM completely, crash prone
+    // Currently no way to know how much RAM is used by OS and other applications
+    if (prefetched_size_sum >= delegated_max_mem)
     {
       Config::SetCurrent(Config::GFX_HIRES_TEXTURES, false);
 
+      WARN_LOG(VIDEO, "Prefetcher: Warning, custom textures prefetching after %.3f MB aborted."
+                      " Maximum delegated memory limit of %.3f breached.",
+               prefetched_size_sum / CNV_BYT_TO_MB, delegated_max_mem / CNV_BYT_TO_MB);
       OSD::AddMessage(
           StringFromFormat(
               "Custom Textures prefetching after %.1f MB aborted, not enough RAM available",
-              size_sum / (1024.0 * 1024.0)),
-          10000);
+              prefetched_size_sum / CNV_BYT_TO_MB),
+          20000);
       return;
     }
-  }
-  u32 stoptime = Common::Timer::GetTimeMs();
-  OSD::AddMessage(StringFromFormat("Custom Textures loaded, %.1f MB in %.1f s",
-                                   size_sum / (1024.0 * 1024.0), (stoptime - starttime) / 1000.0),
-                  10000);
+
+  }  // End Prefetching
+  prefetch_stop_time_total = Common::Timer::GetTimeMs();
+
+  const double pref_sum_total_mb = prefetched_size_sum / CNV_BYT_TO_MB;
+  const double pref_time_total_sec =
+      (prefetch_stop_time_total - prefetch_start_time_total) / CNV_MS_TO_SEC;
+
+  std::string pref_final_msg =
+      StringFromFormat("Prefetcher: Total of %u custom textures loaded, %.1f MB in %.1f seconds.",
```

Comment:
What's the point of this? If someone wants to know how many textures there are in a texture pack, they can just check it in a file browser. The number of textures doesn't in itself have any impact on Dolphin that's noticeable for the user.

Reply:
Does it hurt the performance or stability tho? 

Wasn't the consensus on the OSD messaging, logging and per-type reporting. And now not even this?

Reply:
It doesn't impact stability, and the performance impact is negligible. But we don't add things to Dolphin just for the sake of adding things to Dolphin, we do it if the addition would make Dolphin better for the user.

Reply:
I am a dolphing user. It will make Dolphing better for me.

Reply:
Keep in mind that the improvement for you can be outweighed by the clutter it adds for all the other users. (Yes, the clutter isn't especially large for this feature, but it gets multiplied over very many people.)

Reply:
So if it comes down to being outnumbered, shouldn't there be an extra category of "the people who contribute at least get something in return here or there", as in, an opportunity to test if infact the users would have an issue with it in practice, if not, it can stay in as long as other contributors are at least neutral, otherwise it would get removed.

This is how I see it:
Fellow contributors that don't (maybe they will, let give some time) raise a flag over this are to be considered either in-agreement or neutral. Correct?

Secondly, you are taking the stand of the user base, but that's an assumption. Why don't you give your own opinion. I think that your position is neutral deep down. Correct ?

Reply:
> So if it comes down to being outnumbered, shouldn't there be an extra category of "the people who contribute at least get something in return here or there"

No.

> Fellow contributors that don't (maybe they will, let give some time) raise a flag over this are to be considered either in-agreement or neutral. Correct?

Consider them neutral if you want to, but you can't assume that they're in agreement.

> Secondly, you are taking the stand of the user base, but that's an assumption. Why don't you give your own opinion. I think that your position is neutral deep down. Correct ?

If you want my own opinion, it's actually that I hate this and think this PR should be closed.